1-org/modules/centralized-logging/main.tf

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  value_first_resource  = values(var.resources)[0]
+  logbucket_sink_member = { for k, v in var.resources : k => v if k != var.logging_project_key }
+
+  all_sinks = merge(
+    var.bigquery_options == null ? {} : {
+      for k, v in var.resources : "${v}_bgq" => var.bigquery_options
+    },
+    var.logbucket_options == null ? {} : {
+      for k, v in var.resources : "${v}_lbk" => var.logbucket_options
+    },
+    var.pubsub_options == null ? {} : {
+      for k, v in var.resources : "${v}_pub" => var.pubsub_options
+    },
+    var.storage_options == null ? {} : {
+      for k, v in var.resources : "${v}_sto" => var.storage_options
+    },
+  )
+
+  logging_sink_name_map = {
+    bgq = try("ds-logs-${var.logging_destination_project_id}", "ds-logs")
+    pub = try("tp-logs-${var.logging_destination_project_id}", "tp-logs")
+    sto = try("bkt-logs-${var.logging_destination_project_id}", "bkt-logs")
+    lbk = try("logbkt-logs-${var.logging_destination_project_id}", "logbkt-logs")
+  }
+  destination_uri_map = {
+    bgq = try(module.destination_bigquery[0].destination_uri, "")
+    pub = try(module.destination_pubsub[0].destination_uri, "")
+    sto = try(module.destination_storage[0].destination_uri, "")
+    lbk = try(module.destination_logbucket[0].destination_uri, "")
+  }
+  logging_target_name_prefix = {
+    bgq = "ds_logs_"
+    pub = "tp-logs-"
+    sto = try("bkt-logs-${var.logging_destination_project_id}-", "bkt-logs-")
+    lbk = "logbkt-logs-"
+  }
+
+  part_tables = {
+    false = { use_partitioned_tables = false }
+    true  = { use_partitioned_tables = true }
+  }
+
+  bgq_options_part_tables = var.bigquery_options == null ? local.part_tables.false : !contains(keys(var.bigquery_options), "partitioned_tables") ? local.part_tables.false : var.bigquery_options.partitioned_tables == "true" ? local.part_tables.true : local.part_tables.false
```

Comment:
Lets break this into multiline for readability, a comment will also be useful.

Reply:
Done